src/device/usbd_pvt.h

```diff
@@ -33,6 +33,29 @@
  extern "C" {
 #endif
 
+//--------------------------------------------------------------------+
+// Class Drivers
+//--------------------------------------------------------------------+
+
+typedef struct
+{
+#if CFG_TUSB_DEBUG >= 2
+  char const* name;
+#endif
+
+  void (* init             ) (void);
+  void (* reset            ) (uint8_t rhport);
+  bool (* open             ) (uint8_t rhport, tusb_desc_interface_t const * desc_intf, uint16_t* p_length);
+  bool (* control_request  ) (uint8_t rhport, tusb_control_request_t const * request);
+  bool (* control_complete ) (uint8_t rhport, tusb_control_request_t const * request);
+  bool (* xfer_cb          ) (uint8_t rhport, uint8_t ep_addr, xfer_result_t event, uint32_t xferred_bytes);
+  void (* sof              ) (uint8_t rhport); /* optional */
+} usbd_class_driver_t;
+
+// Invoked when initializing device stack to get additional class drivers.
+// Can optionally implemented by application to extend/overwrite class driver support.
```

Comment:
I think we need a comment that the array must remain in RAM indefinitely?

Reply:
> Everything looks OK (except for adding a bit more documentation).

Yeah, I am always behind in documentation. I will try to do a better job with docs.
 
> I'd still lean towards only maintaining the driver list in the app (instead of dynamically modifying it with the config header file):

I want application to only focus on its extended driver, and not have to construct other class drivers, and I don't like to expose those built-in classes either. App could miss-calculated the driver struct and all the good old built-in stop working, nah not a chance. 
 
> Without the static list in usbd, the usbd.c code would be slightly shorter, and it also allows the linker to optimize out drivers not explicitly listed by the app without the need to modify the tusb_config header (though maybe that's a bad idea, since all the class drivers would need to be built even if only one was used).
> 
> On the other hand, it would break existing code.

It is not that much, especially when the callback is not implemented, the code is still pretty much the same after optimization. I would care less about the code size. One unique array and 2 virtual ones are the same to me. 

Yeah, the app driver should be long-lived, most of the time it is on ROM, I will add it on the next push